components/restapi/core/accounts.go

```diff
@@ -31,33 +37,43 @@ func congestionForAccountID(c echo.Context) (*apimodels.CongestionResponse, erro
 
 	return &apimodels.CongestionResponse{
 		SlotIndex:            slotIndex,
-		Ready:                false, // TODO: update after scheduler is implemented
-		ReferenceManaCost:    0,     // TODO: update after RMC is implemented
-		BlockIssuanceCredits: account.Credits.Value,
+		Ready:                deps.Protocol.MainEngineInstance().Scheduler.IsBlockIssuerReady(accountID),
+		ReferenceManaCost:    0, // TODO: update after RMC is implemented
+		BlockIssuanceCredits: acc.Credits.Value,
 	}, nil
 }
 
-func staking() (*apimodels.AccountStakingListResponse, error) {
-	resp := &apimodels.AccountStakingListResponse{
-		Stakers: make([]*apimodels.ValidatorResponse, 0),
-	}
+func staking(c echo.Context) (*apimodels.AccountStakingListResponse, error) {
+	pageSize, _ := httpserver.ParseUint32QueryParam(c, restapipkg.QueryParameterPageSize)
+	requestedSlotIndex, cursorIndex, _ := httpserver.ParseCursorQueryParam(c, restapipkg.QueryParameterCursor)
 	latestCommittedSlot := deps.Protocol.SyncManager.LatestCommitment().Index()
-	nextEpoch := deps.Protocol.APIForSlot(latestCommittedSlot).TimeProvider().EpochFromSlot(latestCommittedSlot) + 1
 
-	activeValidators, err := deps.Protocol.MainEngineInstance().SybilProtection.EligibleValidators(nextEpoch)
-	if err != nil {
-		return nil, err
+	if cursorIndex == 0 {
+		requestedSlotIndex = latestCommittedSlot
```

Comment:
Why is `requestedSlotIndex` set to `latestCommittedSlot` if `cursorIndex` == 0?
Or how would the request be if requesting the first page of a specific slot index?

Reply:
the cursor index was supposed to be 0 for the first page, that's why I set the requestAtSlotIndex to latestCommittedSlot (that represent current slot and snapshot responses based on it)
But cursor is now string, I could just check for err to see if it was provided. I change also last page response.